src/Lecture2.hs

```diff
@@ -207,8 +249,19 @@ The algorithm of merge sort is the following:
 >>> mergeSort [3, 1, 2]
 [1,2,3]
 -}
+
+half :: [a] -> ([a], [a])
+half xs =
+    ((take s xs), drop s xs)
+    where
+        s = (length xs ) `div` 2
+
 mergeSort :: [Int] -> [Int]
-mergeSort = error "TODO"
+mergeSort l
+  | null l = []
+  | length l == 1 = [head l]
+  | length l == 2 = if head l < head (tail l) then [head l, head $ tail l] else [head $ tail l, head l]
```

Comment:
`length` is very inefficient function. You can avoid using it and pattern match on lists of the corresponding size instead